src/marketplace/ListingsView.js

```diff
@@ -32,117 +38,171 @@ class ListingsView extends React.Component<Props, State> {
-			this.setState({itemsForSale: sales.filter(sale => 
-				allComponents.includes(sale.name)
-				&& getComponentType(verifyComponent(sale.name)) === this.props.sellerType
-			)}); 
-		});
-	}
+			this.setState({
+				chassisForSale: sales.filter(sale => allComponents.includes(sale.name) && getComponentType(verifyComponent(sale.name)) === 'chassis'),
+				weaponsForSale: sales.filter(sale => allComponents.includes(sale.name) && getComponentType(verifyComponent(sale.name)) === 'weapon'),
+				scannersForSale: sales.filter(sale => allComponents.includes(sale.name) && getComponentType(verifyComponent(sale.name)) === 'scanner'),
+				scannerAddonsForSale: sales.filter(sale => allComponents.includes(sale.name) && getComponentType(verifyComponent(sale.name)) === 'scannerAddon'),
+				jammersForSale: sales.filter(sale => allComponents.includes(sale.name) && getComponentType(verifyComponent(sale.name)) === 'jammer'),
+				treadsForSale: sales.filter(sale => allComponents.includes(sale.name) && getComponentType(verifyComponent(sale.name)) === 'treads'),
+				itemsForSale: sales.filter(sale => allComponents.includes(sale.name) && getComponentType(verifyComponent(sale.name)) === 'item')
+			});
 
-	//This function uses the users id and gets the tanks that are active in the marketplace
-	getMarketSalesForTanks(): void  {
-		getMarketTanks(this.state.userId, sales => {
-				this.setState({itemsForSale: sales.filter(sale => !(allComponents.includes(sale.name)))}); 
-				this.getTanksToShow();
+			// After getting the sales, reset the cards according to the sellerType.
+			switch (this.props.sellerType) {
+				case 'chassis':
+					this.setState({cards: this.createCards(this.state.chassisForSale, 'chassis')});
+					break;
+				case 'weapon':
+					this.setState({cards: this.createCards(this.state.weaponsForSale, 'weapon')});
+					break;
+				case 'scanner':
+					this.setState({cards: this.createCards(this.state.scannersForSale, 'scanner')});
+					break;
+				case 'scannerAddon':
+					this.setState({cards: this.createCards(this.state.scannerAddonsForSale, 'scannerAddon')});
+					break;
+				case 'jammer':
+					this.setState({cards: this.createCards(this.state.jammersForSale, 'jammer')});
+					break;
+				case 'treads':
+					this.setState({cards: this.createCards(this.state.treadsForSale, 'treads')});
+					break;
+				case 'item':
+					this.setState({cards: this.createCards(this.state.itemsForSale, 'item')});
+					break;
+				default:
+					break;
+			}
 		});
 	}
 
-	//This function uses tanks id from the state , creates those tanks , and adds them to the array of tanks
-	getTanksToShow() { 
-		const arrayOfTankIds = [];
-		for(let i = 0; i < this.state.itemsForSale.length; i++) {
-			if(this.state.itemsForSale[i].tankId == null) {
+	// Converts SaleObject to Tank.
+	convertSalesToTanks(saleTanks: Array<SaleObject>): void {
+		// Do not convert if there are no tanks to convert.
+		if (saleTanks.length === 0) {
```

Comment:
why do we need this special case? If our code doesn't work when there are no tanks to convert it seems like we have messy logic somewhere...